include/sensors/core/message_handlers/sensors.hpp

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include "can/core/ids.hpp"
+#include "can/core/messages.hpp"
+#include "sensors/core/utils.hpp"
+
+namespace sensor_message_handler {
+
+template <sensor_task_utils::TaskClient SensorTaskClient>
+class SensorHandler {
+  public:
+    explicit SensorHandler(SensorTaskClient &client) : client(client) {}
+    SensorHandler(const SensorHandler &) = delete;
+    SensorHandler(const SensorHandler &&) = delete;
+    auto operator=(const SensorHandler &) -> SensorHandler & = delete;
+    auto operator=(const SensorHandler &&) -> SensorHandler && = delete;
+    ~SensorHandler() = default;
+
+    void handle(sensor_task_utils::TaskMessage &m) {
+        std::visit([this](auto o) { this->visit(o); }, m);
+    }
+
+  private:
+    SensorTaskClient &client;
+
+    void visit(std::monostate &m) {}
+
+    void visit(can_messages::BaselineSensorRequest &m) {
+        send_to_queue(can_ids::SensorType(m.sensor), m);
+    }
+
+    void visit(can_messages::WriteToSensorRequest &m) {
+        send_to_queue(can_ids::SensorType(m.sensor), m);
+    }
+
+    void visit(can_messages::ReadFromSensorRequest &m) {
+        send_to_queue(can_ids::SensorType(m.sensor), m);
+    }
+
+    void send_to_queue(can_ids::SensorType type,
+                       sensor_task_utils::TaskMessage m) {
+        switch (type) {
+            case can_ids::SensorType::temperature: {
+                client.send_humidity_queue(m);
```

Comment:
Is this really supposed to be `send_humidity_queue`?

Reply:
Oh wait this is a combo humidity/temp sensor isn't it

Reply:
yes it is. I'll change the naming to make it less confusing, sorry!